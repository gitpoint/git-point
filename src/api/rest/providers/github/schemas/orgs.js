import { schema } from 'normalizr';
import { initSchema, toTimestamp } from 'utils';

export const orgSchema = new schema.Entity(
  'orgs',
  {},
  {
    idAttribute: org => org.login.toLowerCase(),
    processStrategy: entity => {
      const processed = initSchema();

      // These are provided in both mini & full modes
      processed.id = entity.login.toLowerCase(); // id should be always used for navigation
      processed.login = entity.login;
      processed.avatarUrl = entity.avatar_url;

      // name is only present in full mode, we base our full parsing on its presence
      if (typeof entity.name !== 'undefined') {
        processed.name = entity.name;
        processed.webSite = entity.blog;
        processed.location = entity.location;
        processed.description = entity.description;

        processed.countPublicRepos = entity.public_repos;
        processed.countPrivateRepos = 0;
        processed.countRepos = entity.public_repos;

        processed._entityUrl = entity.html_url;

        processed.createdAt = toTimestamp(entity.created_at);
        processed.updatedAt = toTimestamp(entity.updated_at);

        // Clear avatar cached URL to make sure picture is refetched on profile change
        processed.avatarUrl += `&updatedAt=${processed.updatedAt}`;

        // The entity is to be considered complete.
        processed._isComplete = true;

        if (typeof entity.total_private_repos !== 'undefined') {
          // This org belongs to the authenticated user, update some props
          processed._isAuth = true;
          processed.countPrivateRepos = entity.total_private_repos;
          processed.countRepos += entity.total_private_repos;
        }
      } else {
        // We can try our best to fill in some props on our own:
        processed._entityUrl = `https://github.com/${processed.id}`;
      }

      return processed;
    },
  }
);
