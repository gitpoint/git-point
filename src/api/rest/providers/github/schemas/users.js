import { schema } from 'normalizr';
import { initSchema, toTimestamp } from 'utils';

export const userSchema = new schema.Entity(
  'users',
  {},
  {
    idAttribute: user => user.login.toLowerCase(),
    processStrategy: entity => {
      const processed = initSchema();

      // These are provided in both mini & full modes
      processed.id = entity.login.toLowerCase(); // id should be always used for navigation
      processed.login = entity.login;
      processed.avatarUrl = entity.avatar_url;

      // name is only present in full mode, we base our full parsing on its presence
      if (typeof entity.name !== 'undefined') {
        processed.fullName = entity.name;
        processed.company = entity.company;
        processed.webSite = entity.blog;
        processed.location = entity.location;
        processed.bio = entity.bio;

        processed.countPublicRepos = entity.public_repos;
        processed.countPrivateRepos = 0;
        processed.countRepos = entity.public_repos;
        processed.countFollowers = entity.followers;
        processed.countFollowing = entity.following;

        processed.createdAt = toTimestamp(entity.created_at); // as unix timestamp
        processed.updatedAt = toTimestamp(entity.updated_at); // as unix timestamp

        // Clear avatar cached URL to make sure picture is refetched on profile change
        processed.avatarUrl += `&updatedAt=${processed.updatedAt}`;

        processed._entityUrl = `https://github.com/${entity.login}`;

        // The entity is to be considered complete.
        processed._isComplete = true;
      }

      return processed;
    },
  }
);
