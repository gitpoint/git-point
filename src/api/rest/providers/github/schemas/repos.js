import { schema } from 'normalizr';
import moment from 'moment/min/moment.min';

import { userSchema } from './users';
import { orgSchema } from './orgs';

export const repoSchema = new schema.Entity(
  'repos',
  {
    userOwner: userSchema,
    orgOwner: orgSchema,
  },
  {
    idAttribute: repo => repo.full_name.toLowerCase(),
    processStrategy: entity => {
      const processed = {};

      // These flags should be in all our schemas.
      processed._isComplete = false; // entity not fully fetched yet
      processed._isAuth = false; // entity doesn't belong to the auth user
      processed._entityUrl = entity.html_url; // The github url for the entity. To be used in openInBrowser()
      processed._fetchedAt = moment().format('X');

      processed.id = entity.full_name;
      processed.fullName = entity.full_name;
      processed.shortName = entity.name;
      processed.description = entity.description;
      processed.private = entity.private;
      processed.defaultBranch = entity.default_branch;
      processed.language = entity.language; // needs to be normalized

      if (entity.owner.type === 'User') {
        processed.userOwner = entity.owner;
        processed.orgOwner = false;
      } else {
        processed.userOwner = false;
        processed.orgOwner = entity.owner;
      }

      processed.countStargazzers = entity.stargazers_count;
      processed.countForks = entity.forks_count;
      processed.countWatchers = entity.watchers_count;
      processed.countOpenIssues = entity.open_issues_count;

      processed.hasIssues = entity.has_issues;

      processed._entityUrl = entity.html_url;

      /*
      // These are provided in both mini & full modes
      processed.id = entity.login; // id should be always used for navigation
      processed.login = entity.login;
      processed.avatarUrl = entity.avatar_url;
      processed.description = entity.description;

      // These flags should be in all our schemas.
      processed._isComplete = false; // entity not fully fetched yet
      processed._isAuth = false; // entity doesn't belong to the auth user
      processed._entityUrl = false; // The github url for the entity. To be used in openInBrowser()

      // name is only present in full mode, we base our full parsing on its presence
      if (typeof entity.name !== 'undefined') {
        processed.name = entity.name;
        processed.webSite = entity.blog;
        processed.location = entity.location;

        processed.countPublicRepos = entity.public_repos;
        processed.countPrivateRepos = 0;
        processed.countRepos = entity.public_repos;

        processed._entityUrl = entity.html_url;

        processed.since = moment(entity.created_at).format('X'); // as unix timestamp

        // The entity is to be considered complete.
        processed._isComplete = true;

        if (typeof entity.total_private_repos !== 'undefined') {
          // This org belongs to the authenticated user, update some props
          processed._isAuth = true;
          processed.countPrivateRepos = entity.total_private_repos;
          processed.countRepos += entity.total_private_repos;
        }
      } else {
        // We can try our best to fill in some props on our own:
        processed._entityUrl = `https://github.com/${processed.id}`;
      } */

      return processed;
    },
  }
);

/*

 // GITHUB REPOS SCHEMAS

const fullMember =  { id: 93332398,
    name: 'git-point-site',
    full_name: 'gitpoint/git-point-site',
    owner:
     { login: 'gitpoint',
       id: 30082377,
       avatar_url: 'https://avatars0.githubusercontent.com/u/30082377?v=4',
       gravatar_id: '',
       url: 'https://api.github.com/users/gitpoint',
       html_url: 'https://github.com/gitpoint',
       followers_url: 'https://api.github.com/users/gitpoint/followers',
       following_url: 'https://api.github.com/users/gitpoint/following{/other_user}',
       gists_url: 'https://api.github.com/users/gitpoint/gists{/gist_id}',
       starred_url: 'https://api.github.com/users/gitpoint/starred{/owner}{/repo}',
       subscriptions_url: 'https://api.github.com/users/gitpoint/subscriptions',
       organizations_url: 'https://api.github.com/users/gitpoint/orgs',
       repos_url: 'https://api.github.com/users/gitpoint/repos',
       events_url: 'https://api.github.com/users/gitpoint/events{/privacy}',
       received_events_url: 'https://api.github.com/users/gitpoint/received_events',
       type: 'Organization',
       site_admin: false },
    private: false,
    html_url: 'https://github.com/gitpoint/git-point-site',
    description: null,
    fork: false,
    url: 'https://api.github.com/repos/gitpoint/git-point-site',
    forks_url: 'https://api.github.com/repos/gitpoint/git-point-site/forks',
    keys_url: 'https://api.github.com/repos/gitpoint/git-point-site/keys{/key_id}',
    collaborators_url: 'https://api.github.com/repos/gitpoint/git-point-site/collaborators{/collaborator}',
    teams_url: 'https://api.github.com/repos/gitpoint/git-point-site/teams',
    hooks_url: 'https://api.github.com/repos/gitpoint/git-point-site/hooks',
    issue_events_url: 'https://api.github.com/repos/gitpoint/git-point-site/issues/events{/number}',
    events_url: 'https://api.github.com/repos/gitpoint/git-point-site/events',
    assignees_url: 'https://api.github.com/repos/gitpoint/git-point-site/assignees{/user}',
    branches_url: 'https://api.github.com/repos/gitpoint/git-point-site/branches{/branch}',
    tags_url: 'https://api.github.com/repos/gitpoint/git-point-site/tags',
    blobs_url: 'https://api.github.com/repos/gitpoint/git-point-site/git/blobs{/sha}',
    git_tags_url: 'https://api.github.com/repos/gitpoint/git-point-site/git/tags{/sha}',
    git_refs_url: 'https://api.github.com/repos/gitpoint/git-point-site/git/refs{/sha}',
    trees_url: 'https://api.github.com/repos/gitpoint/git-point-site/git/trees{/sha}',
    statuses_url: 'https://api.github.com/repos/gitpoint/git-point-site/statuses/{sha}',
    languages_url: 'https://api.github.com/repos/gitpoint/git-point-site/languages',
    stargazers_url: 'https://api.github.com/repos/gitpoint/git-point-site/stargazers',
    contributors_url: 'https://api.github.com/repos/gitpoint/git-point-site/contributors',
    subscribers_url: 'https://api.github.com/repos/gitpoint/git-point-site/subscribers',
    subscription_url: 'https://api.github.com/repos/gitpoint/git-point-site/subscription',
    commits_url: 'https://api.github.com/repos/gitpoint/git-point-site/commits{/sha}',
    git_commits_url: 'https://api.github.com/repos/gitpoint/git-point-site/git/commits{/sha}',
    comments_url: 'https://api.github.com/repos/gitpoint/git-point-site/comments{/number}',
    issue_comment_url: 'https://api.github.com/repos/gitpoint/git-point-site/issues/comments{/number}',
    contents_url: 'https://api.github.com/repos/gitpoint/git-point-site/contents/{+path}',
    compare_url: 'https://api.github.com/repos/gitpoint/git-point-site/compare/{base}...{head}',
    merges_url: 'https://api.github.com/repos/gitpoint/git-point-site/merges',
    archive_url: 'https://api.github.com/repos/gitpoint/git-point-site/{archive_format}{/ref}',
    downloads_url: 'https://api.github.com/repos/gitpoint/git-point-site/downloads',
    issues_url: 'https://api.github.com/repos/gitpoint/git-point-site/issues{/number}',
    pulls_url: 'https://api.github.com/repos/gitpoint/git-point-site/pulls{/number}',
    milestones_url: 'https://api.github.com/repos/gitpoint/git-point-site/milestones{/number}',
    notifications_url: 'https://api.github.com/repos/gitpoint/git-point-site/notifications{?since,all,participating}',
    labels_url: 'https://api.github.com/repos/gitpoint/git-point-site/labels{/name}',
    releases_url: 'https://api.github.com/repos/gitpoint/git-point-site/releases{/id}',
    deployments_url: 'https://api.github.com/repos/gitpoint/git-point-site/deployments',
    created_at: '2017-06-04T18:11:50Z',
    updated_at: '2017-09-28T22:13:04Z',
    pushed_at: '2017-10-04T02:08:59Z',
    git_url: 'git://github.com/gitpoint/git-point-site.git',
    ssh_url: 'git@github.com:gitpoint/git-point-site.git',
    clone_url: 'https://github.com/gitpoint/git-point-site.git',
    svn_url: 'https://github.com/gitpoint/git-point-site',
    homepage: null,
    size: 4656,
    stargazers_count: 9,
    watchers_count: 9,
    language: 'CSS',
    has_issues: true,
    has_projects: true,
    has_downloads: true,
    has_wiki: true,
    has_pages: false,
    forks_count: 3,
    mirror_url: null,
    open_issues_count: 0,
    forks: 3,
    open_issues: 0,
    watchers: 9,
    default_branch: 'master',

    permissions: { admin: false, push: true, pull: true } } ];

    // only in full
parent: {},
source: {},
network_count: 9,
subscribers_count: 2

*/
