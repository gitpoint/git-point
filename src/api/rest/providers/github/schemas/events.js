import { schema } from 'normalizr';
import moment from 'moment/min/moment.min';

import { userSchema } from './users';
import { orgSchema } from './orgs';
import { repoSchema } from './repos';

export const eventSchema = new schema.Entity(
  'events',
  {
    actor: userSchema,
    org: orgSchema,
    repo: repoSchema,
  },
  {
    idAttribute: event => event.id,
    processStrategy: entity => {
      const processed = {};

      processed.id = entity.id;
      processed.type = entity.type; // TODO: needs to be normalized in an Enum
      processed.payload = entity.payload; // TODO: needs to be inspected for more nested entities (forkee)
      processed.createdAt = moment(entity.created_at).format('X'); // as unix timestamp

      processed.actor = entity.actor;
      processed.org = entity.org;
      processed.repo = entity.repo;

      // These flags should be in all our schemas.
      processed._isComplete = true; // entity not fully fetched yet
      processed._isAuth = false; // entity doesn't belong to the auth user
      processed._entityUrl = false; // The github url for the entity. To be used in openInBrowser()
      processed._fetchedAt = moment().format('X');

      return processed;
    },
  }
);
