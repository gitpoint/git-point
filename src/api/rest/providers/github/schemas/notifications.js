import { schema } from 'normalizr';
import { initSchema, toTimestamp } from 'utils';

import { repoSchema } from './repos';

export const notificationSchema = new schema.Entity(
  'notifications',
  {
    repo: repoSchema,
  },
  {
    idAttribute: notification => notification.id,
    processStrategy: entity => ({
      ...initSchema(),
      id: entity.id,
      updatedAt: toTimestamp(entity.updated_at),
      unread: entity.unread,
      reason: entity.reason, // TODO: normalize it
      type: entity.subject.type, // TODO: normalize it
      link: entity.subject.url,
      title: entity.subject.title,
      repo: entity.repository,
    }),
  }
);
