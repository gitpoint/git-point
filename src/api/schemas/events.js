import { schema } from 'normalizr';
import { repoSchema } from './repos';

export const eventSchema = new schema.Entity(
  'events',
  {
    repo: repoSchema,
    payload: {
      forkee: repoSchema,
    },
  },
  {
    idAttribute: event => event.id,
  }
);
