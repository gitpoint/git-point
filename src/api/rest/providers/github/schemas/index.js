import { orgSchema } from './orgs';
import { userSchema } from './users';
import { repoSchema } from './repos';
import { eventSchema } from './events';

export default {
  USER: userSchema,
  USER_ARRAY: [userSchema],
  ORG: orgSchema,
  ORG_ARRAY: [orgSchema],
  REPO: repoSchema,
  REPO_ARRAY: [repoSchema],
  EVENT: eventSchema,
  EVENT_ARRAY: [eventSchema],
};
