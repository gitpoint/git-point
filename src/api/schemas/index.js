import { eventSchema } from './events';
import { repoSchema } from './repos';
import { orgSchema } from './orgs';
import { userSchema } from './users';
import { gqlRepoSchema } from './gql-repos';

export default {
  EVENT: eventSchema,
  EVENT_ARRAY: [eventSchema],
  REPO: repoSchema,
  REPO_ARRAY: [repoSchema],
  USER: userSchema,
  USER_ARRAY: [userSchema],
  ORG: orgSchema,
  ORG_ARRAY: [orgSchema],
  GQL_REPO: gqlRepoSchema,
};
