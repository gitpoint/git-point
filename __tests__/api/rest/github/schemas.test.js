import 'react-native';
import { normalize } from 'normalizr';
import schemas from 'api/rest/providers/github/schemas';

import mocks from './mocks';

jest.mock('react-native-i18n', () => {
  return {};
});
Date.now = jest.fn(() => 1508189841664);

it('normalizes user correctly', () => {
  expect(normalize(mocks.userJson, schemas.USER)).toMatchSnapshot();
});

it('normalizes repo correctly', () => {
  expect(normalize(mocks.repoJson, schemas.REPO)).toMatchSnapshot();
});

it('normalizes org correctly', () => {
  expect(normalize(mocks.orgJson, schemas.ORG)).toMatchSnapshot();
});

it('normalizes events correctly', () => {
  expect(normalize(mocks.eventsJson, schemas.EVENT_ARRAY)).toMatchSnapshot();
});

it('normalizes notifications correctly', () => {
  expect(
    normalize(mocks.notificationsJson, schemas.NOTIFICATION_ARRAY)
  ).toMatchSnapshot();
});
