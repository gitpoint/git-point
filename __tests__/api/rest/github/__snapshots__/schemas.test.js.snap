exports[`test normalizes events correctly 1`] = `
Object {
  "entities": Object {
    "events": Object {
      "6723241817": Object {
        "_entityUrl": false,
        "_fetchedAt": 1508189841664,
        "_isAuth": false,
        "_isComplete": false,
        "actor": "machour2",
        "createdAt": 1508176374000,
        "id": "6723241817",
        "org": null,
        "payload": Object {
          "action": "created",
          "comment": Object {
            "author_association": "NONE",
            "body": "bla",
            "created_at": "2017-10-16T17:52:54Z",
            "html_url": "https://github.com/machour/git-point-playground/issues/15#issuecomment-336970090",
            "id": 336970090,
            "issue_url": "https://api.github.com/repos/machour/git-point-playground/issues/15",
            "updated_at": "2017-10-16T17:52:54Z",
            "url": "https://api.github.com/repos/machour/git-point-playground/issues/comments/336970090",
            "user": Object {
              "avatar_url": "https://avatars0.githubusercontent.com/u/32770098?v=4",
              "events_url": "https://api.github.com/users/machour2/events{/privacy}",
              "followers_url": "https://api.github.com/users/machour2/followers",
              "following_url": "https://api.github.com/users/machour2/following{/other_user}",
              "gists_url": "https://api.github.com/users/machour2/gists{/gist_id}",
              "gravatar_id": "",
              "html_url": "https://github.com/machour2",
              "id": 32770098,
              "login": "machour2",
              "organizations_url": "https://api.github.com/users/machour2/orgs",
              "received_events_url": "https://api.github.com/users/machour2/received_events",
              "repos_url": "https://api.github.com/users/machour2/repos",
              "site_admin": false,
              "starred_url": "https://api.github.com/users/machour2/starred{/owner}{/repo}",
              "subscriptions_url": "https://api.github.com/users/machour2/subscriptions",
              "type": "User",
              "url": "https://api.github.com/users/machour2",
            },
          },
          "issue": Object {
            "assignee": null,
            "assignees": Array [],
            "author_association": "NONE",
            "body": "",
            "closed_at": null,
            "comments": 0,
            "comments_url": "https://api.github.com/repos/machour/git-point-playground/issues/15/comments",
            "created_at": "2017-10-16T17:31:03Z",
            "events_url": "https://api.github.com/repos/machour/git-point-playground/issues/15/events",
            "html_url": "https://github.com/machour/git-point-playground/issues/15",
            "id": 265851173,
            "labels": Array [],
            "labels_url": "https://api.github.com/repos/machour/git-point-playground/issues/15/labels{/name}",
            "locked": false,
            "milestone": null,
            "number": 15,
            "repository_url": "https://api.github.com/repos/machour/git-point-playground",
            "state": "open",
            "title": "notif",
            "updated_at": "2017-10-16T17:52:54Z",
            "url": "https://api.github.com/repos/machour/git-point-playground/issues/15",
            "user": Object {
              "avatar_url": "https://avatars0.githubusercontent.com/u/32770098?v=4",
              "events_url": "https://api.github.com/users/machour2/events{/privacy}",
              "followers_url": "https://api.github.com/users/machour2/followers",
              "following_url": "https://api.github.com/users/machour2/following{/other_user}",
              "gists_url": "https://api.github.com/users/machour2/gists{/gist_id}",
              "gravatar_id": "",
              "html_url": "https://github.com/machour2",
              "id": 32770098,
              "login": "machour2",
              "organizations_url": "https://api.github.com/users/machour2/orgs",
              "received_events_url": "https://api.github.com/users/machour2/received_events",
              "repos_url": "https://api.github.com/users/machour2/repos",
              "site_admin": false,
              "starred_url": "https://api.github.com/users/machour2/starred{/owner}{/repo}",
              "subscriptions_url": "https://api.github.com/users/machour2/subscriptions",
              "type": "User",
              "url": "https://api.github.com/users/machour2",
            },
          },
        },
        "repo": "machour/git-point-playground",
        "type": "IssueCommentEvent",
      },
      "6723243870": Object {
        "_entityUrl": false,
        "_fetchedAt": 1508189841664,
        "_isAuth": false,
        "_isComplete": false,
        "actor": "machour2",
        "createdAt": 1508176399000,
        "id": "6723243870",
        "org": null,
        "payload": Object {
          "action": "created",
          "comment": Object {
            "author_association": "NONE",
            "body": "plop",
            "created_at": "2017-10-16T17:53:19Z",
            "html_url": "https://github.com/machour/git-point-playground/issues/14#issuecomment-336970278",
            "id": 336970278,
            "issue_url": "https://api.github.com/repos/machour/git-point-playground/issues/14",
            "updated_at": "2017-10-16T17:53:19Z",
            "url": "https://api.github.com/repos/machour/git-point-playground/issues/comments/336970278",
            "user": Object {
              "avatar_url": "https://avatars0.githubusercontent.com/u/32770098?v=4",
              "events_url": "https://api.github.com/users/machour2/events{/privacy}",
              "followers_url": "https://api.github.com/users/machour2/followers",
              "following_url": "https://api.github.com/users/machour2/following{/other_user}",
              "gists_url": "https://api.github.com/users/machour2/gists{/gist_id}",
              "gravatar_id": "",
              "html_url": "https://github.com/machour2",
              "id": 32770098,
              "login": "machour2",
              "organizations_url": "https://api.github.com/users/machour2/orgs",
              "received_events_url": "https://api.github.com/users/machour2/received_events",
              "repos_url": "https://api.github.com/users/machour2/repos",
              "site_admin": false,
              "starred_url": "https://api.github.com/users/machour2/starred{/owner}{/repo}",
              "subscriptions_url": "https://api.github.com/users/machour2/subscriptions",
              "type": "User",
              "url": "https://api.github.com/users/machour2",
            },
          },
          "issue": Object {
            "assignee": null,
            "assignees": Array [],
            "author_association": "NONE",
            "body": "",
            "closed_at": null,
            "comments": 1,
            "comments_url": "https://api.github.com/repos/machour/git-point-playground/issues/14/comments",
            "created_at": "2017-10-15T14:48:14Z",
            "events_url": "https://api.github.com/repos/machour/git-point-playground/issues/14/events",
            "html_url": "https://github.com/machour/git-point-playground/issues/14",
            "id": 265577462,
            "labels": Array [],
            "labels_url": "https://api.github.com/repos/machour/git-point-playground/issues/14/labels{/name}",
            "locked": false,
            "milestone": null,
            "number": 14,
            "repository_url": "https://api.github.com/repos/machour/git-point-playground",
            "state": "open",
            "title": "more test",
            "updated_at": "2017-10-16T17:53:19Z",
            "url": "https://api.github.com/repos/machour/git-point-playground/issues/14",
            "user": Object {
              "avatar_url": "https://avatars0.githubusercontent.com/u/32770098?v=4",
              "events_url": "https://api.github.com/users/machour2/events{/privacy}",
              "followers_url": "https://api.github.com/users/machour2/followers",
              "following_url": "https://api.github.com/users/machour2/following{/other_user}",
              "gists_url": "https://api.github.com/users/machour2/gists{/gist_id}",
              "gravatar_id": "",
              "html_url": "https://github.com/machour2",
              "id": 32770098,
              "login": "machour2",
              "organizations_url": "https://api.github.com/users/machour2/orgs",
              "received_events_url": "https://api.github.com/users/machour2/received_events",
              "repos_url": "https://api.github.com/users/machour2/repos",
              "site_admin": false,
              "starred_url": "https://api.github.com/users/machour2/starred{/owner}{/repo}",
              "subscriptions_url": "https://api.github.com/users/machour2/subscriptions",
              "type": "User",
              "url": "https://api.github.com/users/machour2",
            },
          },
        },
        "repo": "machour/git-point-playground",
        "type": "IssueCommentEvent",
      },
    },
    "repos": Object {
      "machour/git-point-playground": Object {
        "_entityUrl": "https://github.com/machour/git-point-playground",
        "_fetchedAt": 1508189841664,
        "_isAuth": false,
        "_isComplete": false,
        "fullName": "machour/git-point-playground",
        "id": "machour/git-point-playground",
        "shortName": "git-point-playground",
      },
    },
    "users": Object {
      "machour2": Object {
        "_entityUrl": false,
        "_fetchedAt": 1508189841664,
        "_isAuth": false,
        "_isComplete": false,
        "avatarUrl": "https://avatars.githubusercontent.com/u/32770098?",
        "id": "machour2",
        "login": "machour2",
      },
    },
  },
  "result": Array [
    "6723243870",
    "6723241817",
  ],
}
`;

exports[`test normalizes notifications correctly 1`] = `
Object {
  "entities": Object {
    "notifications": Object {
      "266802681": Object {
        "_entityUrl": false,
        "_fetchedAt": 1508189841664,
        "_isAuth": false,
        "_isComplete": false,
        "id": "266802681",
        "link": "https://api.github.com/repos/machour/git-point-playground/issues/14",
        "reason": "subscribed",
        "repo": "machour/git-point-playground",
        "title": "more test",
        "type": "Issue",
        "unread": true,
        "updatedAt": 1508176400000,
      },
      "267118003": Object {
        "_entityUrl": false,
        "_fetchedAt": 1508189841664,
        "_isAuth": false,
        "_isComplete": false,
        "id": "267118003",
        "link": "https://api.github.com/repos/machour/git-point-playground/issues/15",
        "reason": "subscribed",
        "repo": "machour/git-point-playground",
        "title": "notif",
        "type": "Issue",
        "unread": true,
        "updatedAt": 1508176376000,
      },
    },
    "repos": Object {
      "machour/git-point-playground": Object {
        "_entityUrl": "https://github.com/machour/git-point-playground",
        "_fetchedAt": 1508189841664,
        "_isAuth": false,
        "_isComplete": false,
        "description": null,
        "fullName": "machour/git-point-playground",
        "id": "machour/git-point-playground",
        "orgOwner": false,
        "private": false,
        "shortName": "git-point-playground",
        "userOwner": "machour",
      },
    },
    "users": Object {
      "machour": Object {
        "_entityUrl": false,
        "_fetchedAt": 1508189841664,
        "_isAuth": false,
        "_isComplete": false,
        "avatarUrl": "https://avatars2.githubusercontent.com/u/304450?v=4",
        "id": "machour",
        "login": "machour",
      },
    },
  },
  "result": Array [
    "266802681",
    "267118003",
  ],
}
`;

exports[`test normalizes org correctly 1`] = `
Object {
  "entities": Object {
    "orgs": Object {
      "gitpoint": Object {
        "_entityUrl": "https://github.com/gitpoint",
        "_fetchedAt": 1508189841664,
        "_isAuth": false,
        "_isComplete": true,
        "avatarUrl": "https://avatars0.githubusercontent.com/u/30082377?v=4&updatedAt=1501807857000",
        "countPrivateRepos": 0,
        "countPublicRepos": 2,
        "countRepos": 2,
        "createdAt": 1499788147000,
        "description": "An open source GitHub client for iOS and Android. Built with React Native :iphone:",
        "id": "gitpoint",
        "location": "Toronto",
        "login": "gitpoint",
        "name": "GitPoint",
        "updatedAt": 1501807857000,
        "webSite": "https://gitpoint.co",
      },
    },
  },
  "result": "gitpoint",
}
`;

exports[`test normalizes repo correctly 1`] = `
Object {
  "entities": Object {
    "orgs": Object {
      "gitpoint": Object {
        "_entityUrl": "https://github.com/gitpoint",
        "_fetchedAt": 1508189841664,
        "_isAuth": false,
        "_isComplete": false,
        "avatarUrl": "https://avatars0.githubusercontent.com/u/30082377?v=4",
        "id": "gitpoint",
        "login": "gitpoint",
      },
    },
    "repos": Object {
      "gitpoint/git-point": Object {
        "_entityUrl": "https://github.com/gitpoint/git-point",
        "_fetchedAt": 1508189841664,
        "_isAuth": false,
        "_isComplete": false,
        "countForks": 224,
        "countOpenIssues": 93,
        "countStargazzers": 2501,
        "countWatchers": 2501,
        "defaultBranch": "master",
        "description": "GitHub in your pocket :iphone:",
        "fullName": "gitpoint/git-point",
        "hasIssues": true,
        "id": "gitpoint/git-point",
        "language": "JavaScript",
        "orgOwner": "gitpoint",
        "private": false,
        "shortName": "git-point",
        "userOwner": false,
      },
    },
  },
  "result": "gitpoint/git-point",
}
`;

exports[`test normalizes user correctly 1`] = `
Object {
  "entities": Object {
    "users": Object {
      "machour": Object {
        "_entityUrl": "https://github.com/machour",
        "_fetchedAt": 1508189841664,
        "_isAuth": false,
        "_isComplete": true,
        "avatarUrl": "https://avatars2.githubusercontent.com/u/304450?v=4&updatedAt=1507576478000",
        "bio": null,
        "company": "IDK",
        "countFollowers": 65,
        "countFollowing": 43,
        "countPrivateRepos": 0,
        "countPublicRepos": 56,
        "countRepos": 56,
        "createdAt": 1276477765000,
        "fullName": "Mehdi Achour",
        "id": "machour",
        "location": "Tunis",
        "login": "machour",
        "updatedAt": 1507576478000,
        "webSite": "https://machour.idk.tn/",
      },
    },
  },
  "result": "machour",
}
`;
